extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -32,17 +33,34 @@ export class AnalysesResultsManager {
     void this.logger.log(`Downloading and processing results for ${analysisSummary.nwo}`);
 
     await this.downloadSingleAnalysisResults(analysisSummary, credentials);
+    await publishResults(this.analysesResults);
   }
 
-  public async downloadAllResults(
-    analysisSummaries: AnalysisSummary[],
+  public async downloadAnalysesResults(
+    analysesToDownload: AnalysisSummary[],
+    token: CancellationToken | undefined,
+    publishResults: (analysesResults: AnalysisResults[]) => Promise<void>
   ): Promise<void> {
     const credentials = await Credentials.initialize(this.ctx);
 
-    void this.logger.log('Downloading and processing all results');
+    void this.logger.log('Downloading and processing analyses results');
 
-    for (const analysis of analysisSummaries) {
-      await this.downloadSingleAnalysisResults(analysis, credentials);
+    const batchSize = 3;
+
+    for (let i = 0; i < analysesToDownload.length; i += batchSize) {
+      if (token && token.isCancellationRequested) {
+        throw new Error('Downloading of analyses results has been cancelled');
+      }
+
+      const batch = analysesToDownload.slice(i, i + batchSize);
+      const batchTasks = batch.map(analysis => this.downloadSingleAnalysisResults(analysis, credentials));
+
+      const nwos = batch.map(a => a.nwo).join(', ');
+      void this.logger.log(`Downloading batch ${Math.floor(i / 3) + 1} of analysis results (${nwos})`);
```

Comment:
Can you include the total number of batches somewhere? Either in in this line or before you start downloading?

Maybe something like:

```suggestion
     const numBatches = Math.ceiling(analysesToDownload.length / 3);
      void this.logger.log(`Downloading analysis results batch ${Math.floor(i / 3) + 1} of ${numBatches} (${nwos})`);
```

Reply:
Yep, added.